### PR TITLE
feat (api): git url warnings

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/api/sanity/variable.go [hunk 1]
```diff
@@ -4,25 +4,122 @@ import (
...
-	emap = make(map[string]int)
-	badmap = make(map[string]int)
+//checkGitVariables needs full loaded project, pipeline
+func checkGitVariables(db gorp.SqlExecutor, vars []string, p *sdk.Project, pip *sdk.Pipeline, a *sdk.Action) []sdk.Warning {
+	var warnings []sdk.Warning
+
+	var foundGitURLVar bool
+	for _, v := range vars {
+		if v == "url" {
+			foundGitURLVar = true
```
Comment (yesnault) [inline engine/api/sanity/variable.go]: break

File: engine/api/sanity/variable.go [hunk 2]
```diff
@@ -4,25 +4,122 @@ import (
...
+				break
+			}
+		}
+		//Check key at application level
+		if !hasKey {
+		loopApp:
+			for _, a := range p.Applications {
+				ok, _ := application.IsAttached(db, p.ID, a.ID, pip.Name)
+				if ok {
+					for _, v := range a.Variable {
```
Comment (sguiheux) [inline engine/api/sanity/variable.go]: you don't have application variables here :/

File: engine/api/sanity/variable.go [hunk 3]
```diff
@@ -4,25 +4,122 @@ import (
...
+// - environment variables used in this actions and all its children
+// - git variables used in this actions and all its children
+// - mal formated variables used in this actions and all its children
+
+var (
+	projectVarReg = regexp.MustCompile(`{{\.cds\.proj\.(.*?)}}`)
+	appVarReg     = regexp.MustCompile(`{{\.cds\.app\.(.*?)}}`)
+	envVarReg     = regexp.MustCompile(`{{\.cds\.env\.(.*?)}}`)
+	gitVarReg     = regexp.MustCompile(`{{\.git\.(.*?)}}`)
+	badVarReg     = regexp.MustCompile(`({{cds.*}})|({{ .?cds.*}})|({{git.*}})|({{ .?git.*}})`)
```
Comment (sguiheux) [inline engine/api/sanity/variable.go]: space at the end?

File: engine/api/sanity/variable.go (hunk 1 above)
Comment (fsamin) [inline engine/api/sanity/variable.go]: fixed

File: engine/api/sanity/variable.go (hunk 2 above)
Comment (fsamin) [inline engine/api/sanity/variable.go]: fixed

**Pull request merged**